### DISCUSSION TITLE
How avoid auto parsing result of query to SQLAlchemy model and get raw dict fo speedup?

### DISCUSSION THREAD

**Discussion by shurshilov** [sqlalchemy/sqlalchemy#10661] [Usage Questions] [answered]
![image](https://github.com/sqlalchemy/sqlalchemy/assets/11828278/982d006c-ec92-48f6-82e1-f8c05b6dec6e)
How i can execute query and get raw DICT data? Because I auto get Chain instanse object and i need serialization/deserialization it, and it unnecessary action

**Comment by CaselIT** [comment]
Hi,

You may use a [connection](https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#sqlalchemy.ext.asyncio.AsyncSession.connection). Since that does not have a get method, you can use a select, something like `select(Env.models.chain).where(...)` https://docs.sqlalchemy.org/en/20/tutorial/data_select.html

You get back named tuple like objects, or mapping like object if you use https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#sqlalchemy.ext.asyncio.AsyncResult.mappings

**Reply by shurshilov** [reply]
I try this, but res not simple type((
![image](https://github.com/sqlalchemy/sqlalchemy/assets/11828278/0c4912a4-7139-40f5-81d6-b219fa242864)

**Reply by CaselIT** [reply]
Since you are filtering by id you can use `scalar_one_or_none()` https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#sqlalchemy.ext.asyncio.AsyncResult.scalar_one_or_none
You can learn more here https://docs.sqlalchemy.org/en/20/tutorial/dbapi_transactions.html#tutorial-fetching-rows

also you are still using the session.

Something like this is probably more like what you want
```py
conn = await session.connection()
stmt = select(Env.models.chain).filter_by(id=id)
res = (await conn.execute()).mappings().scalar_one_or_none()
```

**Reply by shurshilov** [reply]
![image](https://github.com/sqlalchemy/sqlalchemy/assets/11828278/c3ddec50-9317-4e8d-8f71-65a39c56f16d)
AttributeError: 'MappingResult' object has no attribute 'scalar_one_or_none'(((

**Reply by gordthompson** [reply, marked as answer]
This (sync) code works for me:

```python
    stmt = select(text("*")).select_from(Subscriber).where(Subscriber.id == 1)
    result = sess.execute(stmt)
    stuff = result.mappings().one_or_none()
    print(stuff)
    # {'id': 1, 'name': 'Gord'}
```

**Reply by shurshilov** [reply]
ITs work
![image](https://github.com/sqlalchemy/sqlalchemy/assets/11828278/001b300b-65b6-475e-bc80-1b712158575e)
But off cause we serializationit again and...it bad(((

**Reply by CaselIT** [reply]
you can keep the select as `select(Env.models.chain).filter_by(id=id)` too.

> But off cause we serializationit again and...it bad(((

not sure that's the error, but if you want an actual dict to serialize you can try doing `res = dict(res)`

**Reply by shurshilov** [reply]
I just want alchemy to return raw data like DictCursor or something like that, the point is that it doesn't even start converting data into orm models, since 50% of my endpoints don't need it, only request and return raw data, thank you

**Reply by zzzeek** [reply]
why are we still using session.execute()?  use session.connection().execute(), you will get tuples / mappings

**Reply by CaselIT** [reply]
> why are we still using session.execute()? use session.connection().execute(), you will get tuples / mappings

oh missed that. sure you need to use the connection as in the example here https://github.com/sqlalchemy/sqlalchemy/discussions/10661#discussioncomment-7623506
